File: sms-sender-java/src/main/java/com/meesho/sms/event/SmsEvent.java
package com.meesho.sms.event;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.OffsetDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class SmsEvent {
    private String eventId;          // stable per-request id (the outbox row id) → consumer dedup key
    private String phoneNumber;
    private String message;
    private String status;
    private String vendorMessageId;
    private String errorReason;
    private OffsetDateTime timestamp;
}

File: sms-store-go/models/sms_event.go
package models

import "time"

type SmsEvent struct {
	EventId         string    `json:"eventId" bson:"eventId"`
	PhoneNumber     string    `json:"phoneNumber" bson:"phoneNumber"`
	Message         string    `json:"message" bson:"message"`
	Status          string    `json:"status" bson:"status"`
	VendorMessageId string    `json:"vendorMessageId" bson:"vendorMessageId"`
	ErrorReason     string    `json:"errorReason" bson:"errorReason"`
	Timestamp       time.Time `json:"timestamp" bson:"timestamp"`
}
